packages/commerce-sdk-react/src/hooks/utils.ts

```diff
@@ -31,6 +31,10 @@ export const updateCache = (queryClient: QueryClient, cacheUpdates: CacheUpdate,
         // If an updater isn't given, fall back to just setting the data
         queryClient.setQueryData(queryKey, updater ?? data)
     )
+
+    if (cacheUpdates.clear) {
+        queryClient.clear()
+    }
```

Comment:
This action intentionally happens after all the others if for some reason there are other updater actions defined alongside of the clear action, which shouldn't ever be the case.